airflow/sensors/base_sensor_operator.py

```diff
@@ -70,6 +70,14 @@ def __init__(self,
                  *args,
                  **kwargs):
         super(BaseSensorOperator, self).__init__(*args, **kwargs)
+        if isinstance(poke_interval, str) or poke_interval < 0:
+             raise AirflowException(
+                "The poke_interval must be a non-negative number"
+                )
+        if isinstance(timeout, str) or timeout < 0:
+             raise AirflowException(
+                "The timeout must be a non-negative number"
+                )
```

Comment:
I'm not sure if it's necessary to check types here. Because if you provide an invalid type for `timeout` or `poke_interval`, the `execute` method will fail and give exception later anyway.

On the other hand, if I provide a Boolean value here for `timeout` or `poke_interval`, it should be captured as invalid type as well while it's not covered here.

Reply:
Thanks @XD-DENG for a quick review, actually sleep(True) is considered as sleep(1) and sleep(0) for sleep(False), so it's also kind of handled already. Thoughts?
I'll remove string type comparison though.

Also, some goof up has been done due to which this PR got closed, I've opened another one https://github.com/apache/airflow/pull/4878